internals-js/src/operations.ts

```diff
@@ -1001,16 +1019,17 @@ export class Operation extends DirectiveTargetElement<Operation> {
       }
     }
 
-    return this.withUpdatedSelectionSetAndFragments(optimizedSelection, finalFragments ?? undefined);
+    return this.withUpdatedSelectionSetAndFragments(optimizedSelection, finalFragments ?? undefined, allAvailableVariables);
   }
 
-  generateQueryFragments(): Operation {
+  generateQueryFragments(allAvailableVariables: VariableDefinitions = new VariableDefinitions()): Operation {
     const [minimizedSelectionSet, fragments] = this.selectionSet.minimizeSelectionSet();
+    
     return new Operation(
       this.schema(),
       this.rootKind,
       minimizedSelectionSet,
-      this.variableDefinitions,
+      allAvailableVariables ? this.collectVariablesFromFragments(allAvailableVariables, fragments) : this.variableDefinitions,
```

Comment:
Since the type of `allAvailableVariables` is `VariableDefinitions`, this ternary will always evaluate to true.